app/atoms/records.ts

```diff
@@ -62,7 +78,7 @@ export const recordsAtom = atom(
         }
       }
 
-      store.setState((s) => ({
+      signal.set((s) => ({
         ...s,
         pendingCellsToFetch: {},
```

Comment:
`setStateDeep` didn't work before for obvious reasons, but would `signal.mutate({ pendingCellsToFetch: {} })` work or does it operate kind of like setStateDeep?

Reply:
Good question. This is actually different depending on the overload used. Your example wouldn't work for the same reasons as `setStateDeep`:

```ts
// recursively mutates for you, but doesn't detect any mutations here:
signal.mutate({ pendingCellsToFetch: {} })
```

```ts
exampleNode.mutate(state => {
  // directly sets `pendingCellsToFetch` to an empty object
  state.pendingCellsToFetch = {}
})
```

Reply:
nice makes sense after explanation - migghttt trip folks up though. setStateDeep made it more clear since it's similar naming to something like mergeDeep which is well understood by most devs. wondering if maybe the overloads could be split, first case is mutateDeep, second case is mutate and only accepts the callback version or something? naming is hard though, and subjective, so i totally get it if you want to stick w the overload 😅

Reply:
It's a fair point. I don't think I'm concerned - I expect the function overload of `.mutate` to be most common. That's how valtio and immer (and all state managers that use it) work, so I imagine it'll be the most familiar.

The object overload would therefore be an "advanced usage for convenience". I already use it a lot, but I imagine many people will never touch it. I'm fine if it has quirks.